ui/Screen/Project/PeerSelector.svelte

```diff
@@ -45,11 +36,13 @@
   };
 
   const dispatch = createEventDispatcher();
-  const selectPeer = peerId => {
+  const selectPeer = (peerId: string) => {
     hideDropdown();
-    currentPeerId = peerId;
     dispatch("select", { peerId });
   };
+
+  $: currentPeer =
+    availablePeers.find(peer => peer.id === currentPeerId) || availablePeers[0];
```

Comment:
Again, what happens if there are no `availablePeers`?

Reply:
The `PeerSelector` wouldn't render in that case - this would mean that proxy didn't return any `Revisions`, or that it returned `Revisions` without an `Identity`, which I don't think it's supposed to do 🤔 .

Alternatively we could throw an error, or display the disabled `PeerSelector` with some sort of 'No peers found" messaging.

Reply:
> which I don't think it's supposed to do .

Don't assume external resources will _always_ do the thing they're supposed to. That's what I'm trying to get at here: we should think about failure cases and show appropriate errors when possible. Indexing arrays is a code smell in my eyes.

Reply:
@xla do you plan to do anything about this in this PR, out of interest?